### DISCUSSION TITLE
A couple of d-vector questions

### DISCUSSION THREAD

**Discussion by Rolun** [coqui-ai/TTS#2982] [General Q&A]
Hello,
I am going through the VITS/YourTTS code and paper and trying to understand a few things about training with d-vectors.

- Are d-vectors computed separately for each audio file instead of averaging them for each speaker? Does this approach work better than using just one d-vector per speaker?
- If we do calculate d-vectors per audio file, why do we use pre-computed ones during training instead of calculating them on the fly (in the training loop)? Especially when we're fine-tuning the speaker encoder, and the speaker embeddings may change for the same audio with each update.
- Why are we calculating a new speaker embedding for the cosine similarity loss rather than using the inputted (pre-computed) d-vector?

Many thanks in advance for any answer to these questions!
